File: Python/Problem_025.py
def solve() -> int:
    a, b = 1, 1  # F1, F2
    idx = 2

    while len(str(b)) < 1000:
        a, b = b, a + b
        idx += 1

    return idx

if __name__ == "__main__":
    print(solve())  # 4782

File: Java/Problem_025.java
import java.math.BigInteger;

public class Main {
    public static void main(String[] args) {
        BigInteger a = BigInteger.ONE; // F1
        BigInteger b = BigInteger.ONE; // F2
        int idx = 2;

        while (b.toString().length() < 1000) {
            BigInteger c = a.add(b);
            a = b;
            b = c;
            idx++;
        }

        System.out.println(idx); // 4782
    }
}
